sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt

```diff
@@ -65,18 +62,19 @@ class CameraActivity : AppCompatActivity() {
             GPUImageFilterTools.showDialog(this) { filter -> switchFilterTo(filter) }
         }
         findViewById<View>(R.id.button_capture).setOnClickListener {
-            when (cameraLoader.cameraInstance!!.parameters.focusMode) {
-                Parameters.FOCUS_MODE_FIXED -> takePicture()
-                Parameters.FOCUS_MODE_CONTINUOUS_PICTURE -> takePicture()
-                else -> cameraLoader.cameraInstance!!.autoFocus { _, _ -> takePicture() }
-            }
+            saveSnapshot()
         }
         findViewById<View>(R.id.img_switch_camera).run {
             setOnClickListener { cameraLoader.switchCamera() }
-            if (!cameraHelper.hasFrontCamera() || !cameraHelper.hasBackCamera()) {
+            if (!cameraLoader.hasMultipleCamera()) {
                 visibility = View.GONE
             }
         }
+        cameraLoader.setOnPreviewFrameListener { data, width, height ->
+            gpuImageView.gpuImage.updatePreviewFrame(data, width, height)
```

Comment:
I'd like to use this method more helpful, so could you make define to GPUImageView too and call that?

Reply:
Done!